### PR TITLE
Next.js 13 Web Vitals & React Server Components

### PR DESCRIPTION
**Pull request by bahlo**
This drops support for Next.js 12 and adds two hooks:

* `useLogger` which returns an Axiom logger
* `useReportWebVitals` which reports web vitals automatically

### CODE REVIEW

**Comment by schehata** [comment]
I've created a webvitals component (not tested yet).

**Comment by schehata** [comment]
finally got the build to work, the repo now uses pnpm with turbo.

**Review by bdsqqq — APPROVED** [review]

File: examples/app-directory/src/app/page.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import styles from './page.module.css';
+import Link from 'next/link';
+import { Logger } from 'next-axiom';
+
+const inter = Inter({ subsets: ['latin'] });
+
+function Home() {
+  const logger = new Logger();
+ 
+  logger.info('Hello from client', { foo: 'bar' });
```
[the remark this answers is not in the document]
Comment (schehata) [reply examples/app-directory/src/app/page.tsx:14]: should we use useEffect in here, or how would you do it?

File: packages/next-axiom-core/src/platform/base.ts [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
+  shouldSendEdgeReport: boolean;
   token: string | undefined;
   environment: string;
   region: string | undefined;
   axiomUrl: string | undefined;
 
   isEnvVarsSet(): boolean;
   getIngestURL(t: EndpointType): string;
-  wrapWebVitalsObject(metrics: NextWebVitalsMetric[]): any;
+  wrapWebVitalsObject(metrics: any[]): any;
```
[the remark this answers is not in the document]
Comment (schehata) [reply packages/next-axiom-core/src/platform/base.ts:16]: the core package doesn't have dependency on nextjs framework (it is the base package for next12 and next13 support). So it doesn't know what is `NextWebVitalsMetric`.

File: packages/next-axiom-core/src/logger.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
-import config, { isVercel, Version } from './config';
+import { config as configurator, isVercel, Version } from './config';
```
[the remark this answers is not in the document]
Comment (schehata) [reply packages/next-axiom-core/src/logger.ts:1]: I think before I had another variable with same name that made a conflict, It doesn't exist now so I revert the alias, nice catch.

File: Contributing.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+👍🎉 First of all, thank you for your interest in next-axiom! We'd love to accept your patches and contributions! 🎉👍
+
+This project accepts contributions. In order to contribute, you should pay attention to a few guidelines:
+
+## Reporting Issues
+
+Bugs, feature requests, and development-related questions should be directed to our GitHub [issue tracker.](https://github.com/axiomhq/next-axiom/issues)
+
+When reporting a bug, please try and provide as much context as possible such as your operating system, NodeJS version and anything else that might be relevant to the bug. For feature requests, please explain what you're trying to do and how the requested feature would help you do that.
```
Comment (bdsqqq) [inline Contributing.md:11]: Can we ask/encourage minimal repro's for bugs alongside the info?

**Review by bdsqqq — APPROVED** [review]
🚢🚢🚢

File: package.json [hunk 5]
```diff
@@ -48,7 +48,7 @@
     "ts-jest": "~29.1",
     "ts-node": "^10.8.0",
     "turbo": "^1.9.9",
-    "typescript": "^4.6.4"
```
Comment (bdsqqq) [inline package.json:49]: ahhh, all the sweeeeeet wins

this avoids a whole host of issues with next-13, see: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-1.html#decoupled-type-checking-between-jsx-elements-and-jsx-tag-types, don't know if we hit them or not but good to know

**Review by bdsqqq — APPROVED** [review]
gotta love up to date ts, very nice

**Comment by schehata** [comment]
Screenshot 2023-07-05 at 7 02 36 PM

This is what the user will see when they update their next12 project to use next-axiom v0.19.0, It's not helpful though.

using pnpm, I was not able to see any peer deps warnings.

**Comment by schehata** [comment]
do we still need to send edge reports or does vercel now send reports for all types of routes?

**Comment by bahlo** [comment]
Closing for https://github.com/axiomhq/next-axiom/pull/134

**Comment by niklasp** [comment]
> Screenshot 2023-07-05 at 7 02 36 PM
> This is what the user will see when they update their next12 project to use next-axiom v0.19.0, It's not helpful though.
> 
> using pnpm, I was not able to see any peer deps warnings.

how can i resolve this? seeing it atm

**Pull request closed without merging**